crates/apps_lib/src/client/rpc.rs

```diff
@@ -1174,6 +1174,35 @@ pub async fn query_rewards<C: namada_sdk::queries::Client + Sync>(
     )
 }
 
+/// Query token total supply.
+pub async fn query_total_supply<N: Namada>(
+    context: &N,
+    args: args::QueryTotalSupply,
+) {
+    let token = args.token;
+    let supply = unwrap_client_response::<N::Client, token::Amount>(
```

Comment:
You can use `get_token_total_supply` from sdk here. Let's also call `format_denominated_amount` on it to print it nicer

Reply:
Do you think it is ok to leave the first part as is? Since we get the right error handling from `unwrap_client_response` (and would have to reimplement it if using `get_token_total_supply`).

Formatted denominated amount in 02c79160c923ab417a4812e4a0f17a9f02022f66